src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/document-[document]/data/+page.svelte

```diff
@@ -13,30 +13,33 @@
     import { collection, type Attributes } from '../../store';
     import { Container } from '$lib/layout';
     import AttributeItem from '../attributeItem.svelte';
-    import { difference, symmetricDifference } from '$lib/helpers/array';
+    import { symmetricDifference } from '$lib/helpers/array';
     import { isRelationship, isRelationshipToMany } from '../attributes/store';
+    import { deepClone } from '$lib/helpers/object';
 
     const databaseId = $page.params.database;
     const collectionId = $page.params.collection;
     const documentId = $page.params.document;
     const editing = true;
 
     const work = writable(
-        Object.keys($doc)
-            .filter((key) => {
-                return ![
-                    '$id',
-                    '$collection',
-                    '$collectionId',
-                    '$databaseId',
-                    '$createdAt',
-                    '$updatedAt'
-                ].includes(key);
-            })
-            .reduce((obj, key) => {
-                obj[key] = $doc[key];
-                return obj;
-            }, {}) as Models.Document
+        deepClone(
+            Object.keys($doc)
+                .filter((key) => {
+                    return ![
+                        '$id',
+                        '$collection',
+                        '$collectionId',
+                        '$databaseId',
+                        '$createdAt',
+                        '$updatedAt'
+                    ].includes(key);
+                })
+                .reduce((obj, key) => {
+                    obj[key] = $doc[key];
+                    return obj;
+                }, {}) as Models.Document
+        )
```

Comment:
I think this function has become quite verbose.

As an alternative, we can create an initWork function, where we have better control flow.

e.g.

```ts
function initWork() {
  let prohibitedKeys = ['$id', ...];
  const filteredKeys = Object.keys($doc)
  // So on and so forth..
}

const work = initWork()
```

Reply:
that's fair. I'll work on this 👍🏻